src/libraries/System.Linq/tests/CountByTests.cs

```diff
@@ -27,6 +27,36 @@ public void CountBy_KeySelectorNull_ThrowsArgumentNullException()
             AssertExtensions.Throws<ArgumentNullException>("keySelector", () => source.CountBy(keySelector, new AnagramEqualityComparer()));
         }
 
+        [Fact]
+        public void CountBy_SourceThrowsOnGetEnumerator()
+        {
+            IEnumerable<int> source = new ThrowsOnGetEnumerator();
+
+            var enumerator = source.CountBy(x => x).GetEnumerator();
```

Comment:
Is this test correct? Isn't this meant to throw when calling `GetEnumerator()`?

Reply:
Yes, the test is correct. The input/source `GetEnumerator()` method is (lazily) called on the `MoveNext()` call on the output enumerator.
The `GetEnumerator()` method of the output enumerator only returns the compiler-generated "YieldIterator" instance which is the same as the returned enumerable (as it implements both IEnumerator* and IEnumerable*)

Reply:
Ah, right. Thanks.